R/asset.R

```diff
@@ -0,0 +1,108 @@
+#' Add an external JavaScript asset as a dygraph dependency
+#'
+#' @param dygraph Dygraph to add asset to
+#' @param type Type of asset (valid options are: 'plugins', 'plotter' or 'dataHandler')
+#' @param name Name of asset
+#' @param path Path to asset JavaScript file
+#' @param version Asset version (e.g. version of package which provides the asset)
+#'
+#' @return A dygraph with the specified asset added.
+#'
+#' @importFrom htmltools htmlDependency
+#'
+#' @export
+dyAsset <- function(dygraph, type, name, path, version = "1.0") {
```

Comment:
We might want to expose the `all.files` parameter (defaulting it to `FALSE`) for cases where users want to easily include an entire directory.